components/brave_wallet/browser/zcash/create_shield_all_transaction_task.cc

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_wallet/browser/zcash/create_shield_all_transaction_task.h"
+
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_wallet/browser/zcash/zcash_serializer.h"
+#include "brave/components/brave_wallet/common/common_utils.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace brave_wallet {
+
+namespace {
+
+std::unique_ptr<OrchardManager> ApplyOrchardSignatures(
+    std::unique_ptr<OrchardManager> orchard_manager,
+    std::array<uint8_t, kZCashDigestSize> sighash) {
```

Comment:
needs dcheck we are not on main thread

Reply:
and a comment that that's a CPU heavy operation

Reply:
Done